dom.bs

```diff
@@ -1186,33 +1186,70 @@ for discussion).
 
  <li><p>Initialize <var>event</var>'s {{Event/currentTarget}} attribute to <var>object</var>.
 
+ <li><p><a>Inner invoke</a> <var>object</var> with <var>event</var>.
+
  <li>
-  <p>For each <var>listener</var> in <var>listeners</var>, whose <b>removed flag</b> is unset, run
-  these substeps:
+  <p>If the return value of <a>inner invoke</a> <var>object</var> with <var>event</var> is false,
+  run these substeps:
 
   <ol>
-   <li><p>If <var>event</var>'s {{Event/type}} attribute value is not <var>listener</var>'s
-   <b>type</b>, terminate these substeps (and run them for the next <a>event listener</a>).
+   <li><p>Set <var>original event type</var> to <var>event</var>'s {{Event/type}} attribute value.
 
-   <li><p>If <var>event</var>'s {{Event/eventPhase}} attribute value is {{Event/CAPTURING_PHASE}}
-   and <var>listener</var>'s <b>capture</b> is false, terminate these substeps (and run them for the
-   next <a>event listener</a>).
+   <li>
+    <p>If <var>event</var>'s {{Event/type}} attribute value is a match for any of the strings in the
+    first column in the following table, set <var>event</var>'s {{Event/type}} attribute value to
+    the string in the second column on the same row as the matching string, otherwise, terminate
+    these substeps.
+
+    <table>
+     <thead>
+      <tr><th>Event type<th>Legacy event type
+     <tbody>
+      <tr><td>"<code>animationend</code>"<td>"<code>webkitAnimationEnd</code>"
+      <tr><td>"<code>animationiteration</code>"<td>"<code>webkitAnimationIteration</code>"
+      <tr><td>"<code>animationstart</code>"<td>"<code>webkitAnimationStart</code>"
+      <tr><td>"<code>transitionend</code>"<td>"<code>webkitTransitionEnd</code>"
+    </table>
+
+    <li><p><a>Inner invoke</a> <var>object</var> with <var>event</var>
+
+    <li><p>Set <var>event</var>'s {{Event/type}} attribute value to <var>original event type</var>.
+  </ol>
+</ol>
 
-   <li><p>If <var>event</var>'s {{Event/eventPhase}} attribute value is {{Event/BUBBLING_PHASE}} and
-   <var>listener</var>'s <b>capture</b> is true, terminate these substeps (and run them for the next
-   <a>event listener</a>).
+<p>To <dfn export for="event listener" id=concept-event-listener-inner-invoke>inner invoke</dfn>
+an <var>object</var> with <var>event</var>, run these steps:
 
-   <li><p>If <var>listener</var>'s <b>passive</b> is true, set <var>event</var>'s
-   <a>in passive listener flag</a>.
+<ol>
+ <li><p>Let <var>listener match</var> be false.
 
-   <li><p>Call <var>listener</var>'s <b>callback</b>'s {{EventListener/handleEvent()}}, with
-   <var>event</var> as argument and <var>event</var>'s {{Event/currentTarget}} attribute value as
-   <a>callback this value</a>. If this throws any exception, <a>report the exception</a>.
+ <li>
+  <p>For each <var>listener</var> in <var>listeners</var>, whose <b>removed flag</b> is unset,
+ run these substeps:
 
-   <li><p>Clear <var>event</var>'s <a>in passive listener flag</a>.
```

Comment:
You dropped this line. We should still have "Unset event's in passive listener flag" after the listener is called.